mbedtls/build/build.rs

```diff
@@ -15,13 +17,27 @@ use rustc_version::Channel;
 // If there's a panic in this code block, that means Cargo's way of running the
 // build script has changed, and this code should be updated to handle the new
 // case.
+// Since Bazel does not use Cargo to generate Crate Metadata hash we need to implement a separate one for it.
 fn get_compilation_metadata_hash() -> String {
     let out_dir: std::path::PathBuf = std::env::var_os("OUT_DIR").unwrap().into();
     let mut out_dir_it = out_dir.iter().rev();
-    assert_eq!(out_dir_it.next().unwrap(), "out");
-    let crate_ = out_dir_it.next().unwrap().to_string_lossy();
-    assert!(crate_.starts_with("mbedtls-"));
-    crate_[8..].to_owned()
+    let next = out_dir_it.next().unwrap();
+
+    if next == "out" {
+        let crate_ = out_dir_it.next().unwrap().to_string_lossy();
+        assert!(crate_.starts_with("mbedtls-"));
+        return crate_[8..].to_owned();
+    } else if next == "_bs.out_dir" {
```

Comment:
And explanation about this string? I mean `"_bs.out_dir"`

Reply:
It's related to the Bazel build system/Rust rule directory structure. Just like Cargo which puts the output in the `out` directory. bazel uses this directory.